staging/src/k8s.io/apiserver/pkg/util/shufflesharding/shufflesharding.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package shufflesharding
+
+import (
+	"fmt"
+	"math"
+)
+
+// MaxHashBits is the max bit length which can be used from hash value.
+// If we use all bits of hash value, the critical(last) value generated
+// by shuffle sharding will be uneven to 2:3 (first half:second half)
+// at most, in order to reduce this unevenness to 32:33, we set MaxHashBits
+// to 60 here.
+const MaxHashBits = 60
+
+// RequiredEntropyBits will validate the entropy requirement for the given
+// deckSize and handSize, will return error when the entropy requirement
+// is more than MaxHashBits. The entropy requirement is evaluated in a
+// fast but approximate way: bits(deckSize^handSize).
+// This must be called before using ShuffleAndDeal.
+func RequiredEntropyBits(deckSize, handSize int) error {
```

Comment:
The name of this function suggests that it returns a number of bits, not an error.  We should fix this mismatch.  There are a few ways we could do this.  One would be to change the return value to a number of bits.  That would be consistent with exposing the limit (MaxHashBits).  These two changes make it the client's responsibility to compare these two quantities and react appropriately.  Alternatively, we could leave the return from this function an `error` and change the name to something like `ValidateEntropyBits`.  But note, the string in the returned error is not exactly what we want in the API validation code.  Perhaps the best option is to make this function simply return a boolean indicating constraint satisfaction or violation, and let the client react as it likes.

Reply:
I was expecting an int.